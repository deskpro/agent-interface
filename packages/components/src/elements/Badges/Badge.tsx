import * as React from "react";
import classNames from "classnames";

import "@deskpro/agent-interface-style/dist/elements/tag-badge.css";

export type BadgeProps = {
  color?: "danger" | "warning" | "warningLight" | "empty" | "yellow" | "white";
};

const Badge: React.SFC<BadgeProps> = ({ children, color }) => (
  <span
    className={classNames("dp-Badge", {
      [`Badge--${color}`]: !!color
    })}
  >
    {children}
  </span>
);

export default Badge;
