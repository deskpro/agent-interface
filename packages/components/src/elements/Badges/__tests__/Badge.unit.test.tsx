import * as React from "react";
import { render, fireEvent } from "react-testing-library";

import Badge from "../Badge";

describe("<Badge />", () => {
  it("should match its snapshot 'id'", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Badge type="id" onClick={handleClick}>
        Badge
      </Badge>
    );
    expect(getByText("Badge")).toMatchSnapshot();
  });
  it("should call onClick fn on click", () => {
    const handleClick = jest.fn();
    const { getByText } = render(
      <Badge type="filterPill" className="test-badge" onClick={handleClick}>
        Badge
      </Badge>
    );

    fireEvent(
      getByText("Badge"),
      new MouseEvent("click", {
        bubbles: true,
        cancelable: true
      })
    );
    expect(handleClick.mock.calls.length).toBe(1);
  });
});
