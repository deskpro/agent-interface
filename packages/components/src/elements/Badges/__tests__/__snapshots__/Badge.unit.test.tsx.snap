// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should match its snapshot 'id' 1`] = `
<span
  class="dp-Badge Badge--id"
  role="button"
  tabindex="-1"
>
  #1234
</span>
`;
